libs/langchain-anthropic/src/chat_models.ts

```diff
@@ -766,6 +782,30 @@ export class ChatAnthropicMessages<
       | Anthropic.Messages.ToolChoiceTool
       | undefined = handleToolChoice(options?.tool_choice);
 
+    if (this.thinking.type === "enabled") {
+      if (this.topK !== -1) {
+        throw new Error("topK is not supported when thinking is enabled");
+      }
+      if (this.topP !== -1) {
+        throw new Error("topP is not supported when thinking is enabled");
+      }
+      if (this.temperature !== 1) {
+        throw new Error(
+          "temperature is not supported when thinking is enabled"
+        );
+      }
+
+      return {
```

Comment:
Do we need a separate return statement here?

Reply:
ha, I was clarifying as you were asking: https://github.com/langchain-ai/langchainjs/pull/7750/files/eba780ee7ff498774ac885a16c7bec271142e51e#r1968553519

Reply:
Ah ok

Maybe we should also ship a minor version that drops the defaults but that can wait